### PR TITLE
Fast confirmation for safe txs including consolidate and optimize txs

### PR DESCRIPTION
**Pull request by gladcow**
This PR adds wallet ability to mark some transactions safe. Outputs from the txes marked as safe can be spent after the first confirmation instead of 10 confirmations for regular transactions. Now simplewallet marks optimize and consolidate txes as safe.

### CODE REVIEW

**Comment by Xecute0101** [comment]
My recommendations are
1. to change the PR title to "Fast confirmation for safe txs including consolidate and optimize txs"
2. change the term "consolidate" to "safe" from commit 5597f6c, since consolidate and optimize are both categorized as safe.

File: lib/Transfers/TransfersSynchronizer.cpp [hunk 1]
```diff
@@ -120,6 +120,13 @@ void TransfersSyncronizer::addPublicKeysSeen(
     }
 }
 
+void TransfersSyncronizer::markTransactionSafe(const Hash &transactionHash)
```
Comment (nnian) [inline lib/Transfers/TransfersSynchronizer.cpp]: The misspellings was already inside the repo, so just as a note:

Syncronizer / Synchronizer not a big deal

**Review by nnian — APPROVED** [review]
Looks good.

command line: arg_fixed_difficulty could be an easy trick for testing in testnet. Well done

**Review by exploshot — APPROVED** [review]

**Review by exploshot — COMMENTED** [review]
reviewed

**Pull request merged**